pkg/sql/resolver.go

```diff
@@ -205,6 +206,31 @@ func (p *planner) CommonLookupFlags(required bool) tree.CommonLookupFlags {
 	}
 }
 
+// IsTableVisible is part of the tree.EvalDatabase interface.
+func (p *planner) IsTableVisible(
+	ctx context.Context, searchPath sessiondata.SearchPath, tableId int64,
+) (bool, error) {
+	tableDesc, err := p.LookupTableByID(ctx, descpb.ID(tableId))
+	if err != nil {
+		return false, err
+	}
+	schemaId := tableDesc.GetParentSchemaID()
+	schemaDesc, err := p.Descriptors().GetImmutableSchemaByID(ctx, p.Txn(), schemaId,
+		tree.SchemaLookupFlags{
+			Required:    true,
+			AvoidCached: p.avoidCachedDescriptors})
+	if err != nil {
+		return false, nil
+	}
+	iter := searchPath.Iter()
+	for scName, ok := iter.Next(); ok; scName, ok = iter.Next() {
```

Comment:
do we need to check database too? what if db is in dropping state?

Reply:
uhh i guess not since it relies on search path.

Reply:
hmm actually my code won't work because you could have two schemas in different databases, but with the same name.

Reply:
i guess the old code had the same bug..

Reply:
HMMM though actually when cross-database references are fully banned, then we don't need to worry about it... (https://github.com/cockroachdb/cockroach/issues/55791)